src/bdecode.cpp

```diff
@@ -697,7 +697,7 @@ namespace libtorrent
 					// we push it into the stack so that we know where to fill
 					// in the next_node field once we pop this node off the stack.
 					// i.e. get to the node following the dictionary in the buffer
-					ret.m_tokens.push_back(bdecode_token(start - orig_start
+					ret.m_tokens.push_back(bdecode_token(std::uint32_t(start - orig_start)
```

Comment:
do you think it's better to make the constructor take `std::size_t` and inside make the cast?